src/neo4j_graphrag/llm/openai_llm.py

```diff
@@ -64,6 +65,32 @@ def get_messages(
             {"role": "system", "content": input},
         ]
 
+    def get_conversation_history(
+        self,
+        input: str,
+        chat_history: list[str],
+    ) -> Iterable[ChatCompletionMessageParam]:
+        messages = [{"role": "system", "content": self.system_instruction}]
+        for i, message in enumerate(chat_history):
+            if i % 2 == 0:
```

Comment:
This assumes that the chat history is a list of strings, alternating between `user` and `assistant` messages. It's obviously fragile and should probably be changed to some kind of `UserMessage` and `AssistantMessage` objects, or smtn like that.

Reply:
Maybe we can use dicts with "role" and "content" keys (it'd work for OpenAI, I don't know about the others)

Reply:
I can look into it.